### PR TITLE
Bootstrapping done. For now.

### PR DESCRIPTION
**Pull request by jforgue**
My text editor died on my last commit so hopefully there's no funky merge issues I missed. Everything looks good for now but there will be functionality changes coming so future style changes are inevitable.

### CODE REVIEW

**Comment by Anbranin** [comment]
Can we get some screenshots? I know you shared them to Slack but they should be here as well.

**Comment by jforgue** [comment]
screen shot 2017-07-25 at 3 34 48 pm
screen shot 2017-07-22 at 6 50 07 pm
screen shot 2017-07-25 at 3 33 12 pm
screen shot 2017-07-25 at 3 33 28 pm
screen shot 2017-07-25 at 3 33 38 pm
screen shot 2017-07-25 at 3 33 48 pm
screen shot 2017-07-25 at 3 33 58 pm
screen shot 2017-07-25 at 3 34 22 pm

File: app/helpers/assignment_index_helper.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+module AssignmentIndexHelper
+  def cal_day(day:)
+    'current-day' if day == Date.current
+  end
+
+  def cal_assignment(assignment:, day:)
+    return if assignment.blank?
+    ass = 'assignment'
```
Comment (mboneil10) [inline app/helpers/assignment_index_helper.rb]: @jforgue, what was the first thing to do on your list? (Hint: it has to do with abbreviations)

Comment (mboneil10) [inline app/helpers/assignment_index_helper.rb]: (This isn't the end of my review, btw. Will look at this again.)

Comment (jforgue) [inline app/helpers/assignment_index_helper.rb]: I thought it would be okay in this case to abbreviate since I cant use assignment and it's only a local variable. But I can certainly change it to something else less, conspicuous?

Comment (mboneil10) [inline app/helpers/assignment_index_helper.rb]: Okay, I don't care about it being inappropriate. However distracting it is for a programmer to read `ass` over and over again. 

All I care about is how well the name represents what it is/does. I figured that `assignment` would not work out. Ummmmm. I'll think of a good alternative if you can't think of something.

Comment (dfaulken) [inline app/helpers/assignment_index_helper.rb]: Maybe something about classes?

**Comment by dfaulken** [comment]
Travis failed because coverage dropped.

**Comment by dfaulken** [comment]
Whoops, didn't mean to close.

File: app/views/assignments/edit.haml [hunk 2]
```diff
@@ -1,12 +1,20 @@
...
-
-= button_to 'Delete assignment', roster_assignment_path(@roster, @assignment),
-  method: :delete
+  %div.container-fluid
+    %div.row
+      %div.col-xs-3
+      %div.col-xs-2
+        %div.form-group
+          = f.label :user_id
+          = f.select :user_id, @users.pluck(:last_name, :id), class: 'form-control'
```
Comment (dfaulken) [inline app/views/assignments/edit.haml]: Can you use `options_from_collection_for_select` here?

File: app/views/assignments/index.haml [hunk 3]
```diff
@@ -1,32 +1,32 @@
...
-                  %span.right &#8592;
-            - else
-              .slightly-red= link_to 'None assigned',
-                new_roster_assignment_url(@roster, date: day)
-  %span.left= link_to @month_date.last_month.strftime('%B %Y'),
+        - assignment = Assignment.in(@roster).on(day)
+        %td.cal-cell{ class: cal_day(day: day) }
+          %div.date
+            = day.strftime '%-d'
+            = if (day.strftime '%-d') == '1'
```
Comment (dfaulken) [inline app/views/assignments/index.haml]: I would put the parentheses around the argument, not the expression

File: app/views/layouts/_nav.haml [hunk 4]
```diff
@@ -0,0 +1,28 @@
+%nav.navbar.navbar-default.nav-custom
+  %div.container-fluid
+    %div.navbar-header
+      %button.navbar-toggle.collapsed{ type: 'button',
+       'data-toggle': 'collapse', 'data-target': '#bs-example-navbar-collapse-1',
```
Comment (dfaulken) [inline app/views/layouts/_nav.haml]: Should use a sub-hash for data

File: app/views/layouts/_nav.haml [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+        'aria-expanded': 'false' }
+
+    %div.collapse.navbar-collapse{ id: 'bs-example-navbar-collapse-1' }
```
Comment (dfaulken) [inline app/views/layouts/_nav.haml]: Should use `#` for static IDs

Also I might suggest using `haml_lint`

File: app/views/layouts/_nav.haml [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+          - if @current_user.admin_in? @roster
+            %li{:class => active_class(roster_users_path(@roster))}
+              = link_to 'Manage users', roster_users_path(@roster)
+            %li{:class => active_class(rotation_generator_roster_assignments_path(@roster))}
+              = link_to 'Generate rotation',
+              rotation_generator_roster_assignments_path(@roster)
+      %ul.nav.navbar-nav.navbar-right
+        %li.name
+          Logged in as #{@current_user.full_name}
+        %li{:class => active_class(edit_roster_user_path(@roster, @current_user))}
```
Comment (dfaulken) [inline app/views/layouts/_nav.haml]: Shouldn't use hashrockets

File: app/helpers/assignment_index_helper.rb (hunk 1 above)
Comment (jforgue) [inline app/helpers/assignment_index_helper.rb]: I took care of it, it'll be in the next commit.

File: app/helpers/assignment_index_helper.rb [hunk 7]
```diff
@@ -7,17 +7,17 @@ def cal_day(day:)
 
   def cal_assignment(assignment:, day:)
     return if assignment.blank?
-    ass = 'assignment'
-    ass += '-user' if assignment.user == @current_user
+    assign = 'assignment'
+    assign += '-user' if assignment.user == @current_user
 
     if day == assignment.start_date && day == assignment.end_date
```
Comment (mboneil10) [inline app/helpers/assignment_index_helper.rb]: Why can't you just say `day == assignment.start_date == assignment.end_date`? Maybe that's more of a mess, but it would remove the need to write `day` twice. Idk. Maybe it's fine.

Comment (mboneil10) [inline app/helpers/assignment_index_helper.rb]: Why would the `start date` and `end date` be the same anyway?

File: app/views/assignments/new.haml [hunk 8]
```diff
@@ -1,19 +1,22 @@
...
+  .container-fluid
+    .row
+      .col-xs-3
+      .col-xs-2
+        .form-group
           = f.label :user_id
-          = f.select :user_id, @users.pluck(:last_name, :id), class: 'form-control'
-      %div.col-xs-4
-        %div.form-group
+          = f.select :user_id, @users.pluck(:last_name, :id),
```
Comment (mboneil10) [inline app/views/assignments/new.haml]: Why isn't this `options_for_select` or whatever it's called?

File: app/helpers/assignment_index_helper.rb (hunk 7 above)
Comment (dfaulken) [inline app/helpers/assignment_index_helper.rb]: I don't think that syntax works. It does if you group the expressions, but that doesn't do what you want:

```ruby
false == false == false # should be true, right? SyntaxError
(false == false) == false # OK, let's try that
true == false # Uh oh
false
```

File: app/views/assignments/new.haml (hunk 8 above)
Comment (dfaulken) [inline app/views/assignments/new.haml]: `options_from_collection_for_select`

Comment (jforgue) [inline app/views/assignments/new.haml]: Sorry missed this one

File: app/helpers/assignment_index_helper.rb (hunk 7 above)
Comment (mboneil10) [inline app/helpers/assignment_index_helper.rb]: Thanks! Good to know. I thought there could be a way to do this, but it might just be better to leave it as is.

File: app/views/assignments/new.haml (hunk 8 above)
Comment (mboneil10) [inline app/views/assignments/new.haml]: What a ridiculously long name. `options_from_collection_for_select`, then.

Comment (jforgue) [inline app/views/assignments/new.haml]: I'll grep this time

File: app/helpers/assignment_index_helper.rb (hunk 7 above)
Comment (dfaulken) [inline app/helpers/assignment_index_helper.rb]: The `start_date` and `end_date` are the same if someone is on call for a single day, e.g. 3pm Friday to 3pm Saturday.

File: app/views/assignments/new.haml (hunk 8 above)
Comment (dfaulken) [inline app/views/assignments/new.haml]: If memory serves, they were introduced in that order. It used to be you had to manually build collections to make your `options_for_select`, but then someone was like "why are we doing this" and so they wrote options *from collection* for select.

File: app/assets/stylesheets/application.scss [hunk 9]
```diff
@@ -27,40 +34,35 @@ body{
...
-  padding: 0.5em 0;
-  background-color: lightyellow;
-  border: 3px solid limegreen;
+.center {
+  text-align: center;
 }
 
-.left{
-  float: left;
+.vague-warning{
```
Comment (dfaulken) [inline app/assets/stylesheets/application.scss]: You crack me up

File: app/assets/stylesheets/application.scss [hunk 10]
```diff
@@ -27,40 +34,35 @@ body{
...
+  background-color: lightpink;
```
Comment (dfaulken) [inline app/assets/stylesheets/application.scss]: 👍 for continuing my unnecessary trend of only using named colors

File: app/assets/stylesheets/assignments/calendar.scss [hunk 11]
```diff
@@ -0,0 +1,73 @@
+#calender {
```
Comment (dfaulken) [inline app/assets/stylesheets/assignments/calendar.scss]: Whoops

File: app/assets/stylesheets/assignments/calendar.scss [hunk 12]
```diff
@@ -0,0 +1,73 @@
...
+}
+
+.cal-head {
+  text-align: right;
+  color: #999;
+  font-weight: normal;
+}
+
+.cal-cell {
+  width: 14%;
```
Comment (dfaulken) [inline app/assets/stylesheets/assignments/calendar.scss]: With weird numbers like this (I assume it's vaguely 1/7) it's nice to explain how you arrived at them.

File: app/assets/stylesheets/assignments/calendar.scss [hunk 13]
```diff
@@ -0,0 +1,73 @@
...
+}
+
+.date {
```
Comment (dfaulken) [inline app/assets/stylesheets/assignments/calendar.scss]: It's up to @Anbranin how much it matters, but some style guides will tell you to alphabetize your CSS properties.

File: app/assets/stylesheets/assignments/calendar.scss [hunk 14]
```diff
@@ -0,0 +1,73 @@
...
+  text-align: right;
+  display: block;
+  height: 0;
+  font-size: 0.8em;
+  padding: 0.25em 0.75em 73% 0.25em;
```
Comment (dfaulken) [inline app/assets/stylesheets/assignments/calendar.scss]: Eesh. Mixing `em` and `%` is okay I guess, but again it'd be nice to know why.

File: app/assets/stylesheets/assignments/calendar.scss [hunk 15]
```diff
@@ -0,0 +1,73 @@
...
+  position: relative;
+  border-top: 1px solid #DEDBDB;
```
Comment (dfaulken) [inline app/assets/stylesheets/assignments/calendar.scss]: Some mixing in this PR of lowercase vs capital hex 11-15. Personally I advocate for lowercase in that capital letters all have the same shape.

File: app/assets/stylesheets/assignments/calendar.scss [hunk 16]
```diff
@@ -0,0 +1,73 @@
...
+.assignment-start {
+  border-bottom-left-radius: 6px;
+  border-top-left-radius: 6px;
+}
+
+.assignment-only {
+  border-radius: 6px;
+}
+
+.assignment {
```
Comment (dfaulken) [inline app/assets/stylesheets/assignments/calendar.scss]: Seems like some nesting of CSS properties could help clean things up in this vague region of the file.

File: app/assets/stylesheets/assignments/calendar.scss [hunk 17]
```diff
@@ -0,0 +1,73 @@
...
+}
+
+.assignment-user {
+  background-color: #ECE9D4;
+}
+
+.width {
+  width: 90%;
+}
+
```
Comment (dfaulken) [inline app/assets/stylesheets/assignments/calendar.scss]: Shouldn't need two newlines here, or newline at EOF

File: app/assets/stylesheets/assignments/index.scss [hunk 18]
```diff
@@ -1,45 +1,18 @@
...
-  background-color: lightpink;
-}
-
-.slightly-green{
-  background-color: lightgreen;
-}
-
-.lightgrey{
-  color: lightgray;
+.space {
```
Comment (dfaulken) [inline app/assets/stylesheets/assignments/index.scss]: Why does it have space

File: app/assets/stylesheets/assignments/calendar.scss [hunk 19]
```diff
@@ -0,0 +1,73 @@
...
+
+.change-month {
```
Comment (dfaulken) [inline app/assets/stylesheets/assignments/calendar.scss]: Also some mixing in this PR of whether you put spaces before an opening curly brace

File: app/controllers/changes_controller.rb [hunk 20]
```diff
@@ -5,7 +5,9 @@ def undo
     version = PaperTrail::Version.find(params.require :id)
     # ... and return is correct here
     # rubocop:disable Style/AndOr
-    head :unauthorized and return unless version.done_by? @current_user
+    head :unauthorized and return unless version
+                                         .whodunnit
+                                         .to_i == @current_user.id
```
Comment (dfaulken) [inline app/controllers/changes_controller.rb]: Kind of awkward to read. I might instead decide if the user IDs match in a separate expression.

File: app/helpers/assignment_index_helper.rb [hunk 21]
```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+module AssignmentIndexHelper
+  def cal_day(day:)
```
Comment (dfaulken) [inline app/helpers/assignment_index_helper.rb]: Just to confirm that it being `nil` is okay when `day != Date.current`?

Also I think there was some discussion of whether `Date.current` is what you actually want here.

File: app/views/assignments/_current.haml [hunk 22]
```diff
@@ -1,11 +1,11 @@
-#current_assignment
+#current-assignment
   - if @current_assignment.present?
-    %h3.slightly-green Currently on call: #{@current_assignment.user.last_name}
+    %h3.vague-success Currently on call: #{@current_assignment.user.last_name}
```
Comment (dfaulken) [inline app/views/assignments/_current.haml]: What's vaguely successful about this message? 😛

File: app/views/assignments/edit.haml [hunk 23]
```diff
@@ -1,12 +1,23 @@
...
-  = f.label :user_id
-  = f.select :user_id, @users.pluck(:last_name, :id)
-  = f.submit 'Save'
-
-= button_to 'Delete assignment', roster_assignment_path(@roster, @assignment),
-  method: :delete
+  .container-fluid
+    .row
+      .col-xs-3
+      .col-xs-2
```
Comment (dfaulken) [inline app/views/assignments/edit.haml]: I think you should be able to ditch the leading `.col-xs-3` in favor of making this `.col-xs-2.col-xs-offset-3`.

File: app/views/assignments/index.haml [hunk 24]
```diff
@@ -1,32 +1,33 @@
...
-                  %span.right &#8592;
-            - else
-              .slightly-red= link_to 'None assigned',
-                new_roster_assignment_url(@roster, date: day)
-  %span.left= link_to @month_date.last_month.strftime('%B %Y'),
+        - assignment = Assignment.in(@roster).on(day)
+        %td.cal-cell{ class: cal_day(day: day) }
+          .date
+            = day.strftime '%-d'
+            = if day.strftime('%-d') == '1'
```
Comment (dfaulken) [inline app/views/assignments/index.haml]: I think here `if day.day == 1` might be clearer, rather than formatting a date as a string in order to compare it to another string.

Perhaps most clear would be `if day == day.month.beginning_of_month`, but that's probably overkill.

File: app/views/assignments/new.haml [hunk 25]
```diff
@@ -1,10 +1,22 @@
...
-  = f.label :end_date
-  = f.text_field :end_date, value: @end_date, class: 'datepicker'
-  = f.label :user_id
-  = f.select :user_id, @users.pluck(:last_name, :id)
-  = f.hidden_field :roster_id, value: @roster.id
-  = f.submit 'Create'
+  .container-fluid
+    .row
+      .col-xs-3
+      .col-xs-2
```
Comment (dfaulken) [inline app/views/assignments/new.haml]: Same here with `offset`

File: app/views/assignments/rotation_generator.haml [hunk 26]
```diff
@@ -1,20 +1,30 @@
...
-  = label_tag :user_ids, 'Users'
-  = select_tag :user_ids,
-    options_from_collection_for_select(@users, :id, :last_name,
-                                       @users.pluck(:id)),
-    multiple: true, size: 10
-  = label_tag :starting_user_id, 'Starting with'
-  = select_tag :starting_user_id,
-    options_from_collection_for_select(@users, :id, :last_name)
-  = submit_tag 'Generate rotation'
+  .container-fluid
```
Comment (dfaulken) [inline app/views/assignments/rotation_generator.haml]: OOC, what's the difference between `.container` and `.container-fluid`?

File: app/views/assignments/rotation_generator.haml [hunk 27]
```diff
@@ -1,20 +1,30 @@
...
+    .row
+      .col-xs-2
+      .col-xs-4
```
Comment (dfaulken) [inline app/views/assignments/rotation_generator.haml]: Same here with `offset`.

File: app/views/layouts/_nav.haml [hunk 28]
```diff
@@ -0,0 +1,30 @@
+%nav.navbar.navbar-default.nav-custom
+  .container-fluid
+    .navbar-header
+      %button.navbar-toggle.collapsed{ type: 'button',
+       data: { toggle: 'collapse', target: '#bs-example-navbar-collapse-1' },
+        aria: { expanded: 'false' } }
+
```
Comment (dfaulken) [inline app/views/layouts/_nav.haml]: Also generally some unnecessary newlines in the HAML in this PR

File: app/views/rosters/new.haml [hunk 29]
```diff
@@ -1,8 +1,16 @@
...
-  To assign the fallback user, first create the roster, then add some users to it.
+  .container-fluid
+    .row
+      .col-xs-4
+      .col-xs-4
+        .form-group
+          = f.label :name
+          = f.text_field :name, class: 'form-control'
+  .center
+    / TODO add functionality so this message isn't necessary
```
Comment (dfaulken) [inline app/views/rosters/new.haml]: Meh, unless we'll make it all one screen: creating a roster, adding users, and assigning a fallback.

File: app/views/twilio/call.haml [hunk 30]
```diff
@@ -1,3 +1,3 @@
 !!! XML
-%Response
-  %Dial= @user.phone
+%response
```
Comment (dfaulken) [inline app/views/twilio/call.haml]: Why this change?

File: app/views/assignments/rotation_generator.haml (hunk 27 above)
Comment (dfaulken) [inline app/views/assignments/rotation_generator.haml]: There are actually quite a few instances, I'm going to stop marking them at this point

File: app/views/users/edit.haml [hunk 31]
```diff
@@ -1,33 +1,4 @@
...
-    = f.check_box :change_notifications_enabled
-    - if @current_user.admin_in? @roster
-      .input-group
-        = f.fields_for @user.membership_in(@roster) do |g|
-          = g.label :admin, "Admin in #{@roster.name}?"
-          = g.check_box :admin
-  .input-group
-    = f.submit 'Save'
+= render 'user', obj: @user, url: roster_user_url(@roster, @user),
+  roster: @user.rosters.pluck(:id), field: @user.membership_in(@roster)
```
Comment (dfaulken) [inline app/views/users/edit.haml]: Any way to cut down on the number of locals? None of the logic is that complicated, and doing anything other than passing one (mayyyybe two) objects makes it so that interpreting the partial involves looking back at how it's invoked a bunch, which isn't ideal.

In particular if you can *just* use the instance variables then you don't need any locals at all.

File: app/views/twilio/call.haml (hunk 30 above)
Comment (dfaulken) [inline app/views/twilio/call.haml]: Not sure if XML tags are case-sensitive, or how permissive Twilio is to this if so, is why I ask.

Comment (werebus) [inline app/views/twilio/call.haml]: I assume haml lint.

But yes, XML is case-sensitive and those tags **must** be capitalized.

Comment (dfaulken) [inline app/views/twilio/call.haml]: Probably this shouldn't be a HAML file huh

File: app/views/assignments/rotation_generator.haml (hunk 27 above)
Comment (jforgue) [inline app/views/assignments/rotation_generator.haml]: I somehow managed to forget offset was an option for a day or two, not sure why but I'll fix it.

File: app/views/layouts/_nav.haml (hunk 28 above)
Comment (jforgue) [inline app/views/layouts/_nav.haml]: Must have picked the wrong haml-lint...

File: app/views/twilio/call.haml (hunk 30 above)
Comment (jforgue) [inline app/views/twilio/call.haml]: Again, I have chosen... poorly.

Comment (dfaulken) [inline app/views/twilio/call.haml]: No, no, I think if I'd given this file a proper extension, perhaps haml lint wouldn't have intervened.

File: app/views/assignments/rotation_generator.haml (hunk 26 above)
Comment (jforgue) [inline app/views/assignments/rotation_generator.haml]: I decided to keep with the theme of scaling and in one particular case .container was causing me some issues.

File: app/views/layouts/_nav.haml (hunk 28 above)
Comment (dfaulken) [inline app/views/layouts/_nav.haml]: I don't think haml lint cares. I mention it more as a point of consistency with the rest of the HAML in this project. I typically don't add newlines in HAML. Personally I find that the indentation provides enough of a visual indication of where sections of the code begin and end.

File: app/views/assignments/rotation_generator.haml (hunk 26 above)
Comment (dfaulken) [inline app/views/assignments/rotation_generator.haml]: OK. But what's the difference?

File: app/assets/stylesheets/assignments/calendar.scss (hunk 11 above)
Comment (jforgue) [inline app/assets/stylesheets/assignments/calendar.scss]: Slowly everyone is noticing that spelling isn't my strong suit.

File: app/assets/stylesheets/assignments/calendar.scss (hunk 12 above)
Comment (jforgue) [inline app/assets/stylesheets/assignments/calendar.scss]: As it turns out I didn't have to be that specific.

File: app/assets/stylesheets/assignments/calendar.scss (hunk 15 above)
Comment (jforgue) [inline app/assets/stylesheets/assignments/calendar.scss]: I originally thought it should be capitalized similar to SQL but have since realized it's rarely capitalized in practice.

File: app/assets/stylesheets/assignments/index.scss (hunk 18 above)
Comment (jforgue) [inline app/assets/stylesheets/assignments/index.scss]: I didn't like how the image went to the edge of the background.

Comment (dfaulken) [inline app/assets/stylesheets/assignments/index.scss]: Okay. I was (unsuccessfully) hinting that it's not a very specific class name

File: app/views/twilio/call.haml (hunk 30 above)
Comment (jforgue) [inline app/views/twilio/call.haml]: Haml lint doesn't seem to notice the xml extension and still points out the "issues".

File: app/assets/stylesheets/assignments/calendar.scss (hunk 11 above)
Comment (mboneil10) [inline app/assets/stylesheets/assignments/calendar.scss]: That's okay, other things can be your strong suit. This is what PRs are for.

File: app/views/twilio/call.haml (hunk 30 above)
Comment (dfaulken) [inline app/views/twilio/call.haml]: Good to know, thanks.

File: app/helpers/assignment_index_helper.rb (hunk 21 above)
Comment (jforgue) [inline app/helpers/assignment_index_helper.rb]: It appears upon inspection that returning nil is okay here. It seems to me that having a time zone defined for the current day will come in handy when the functionality is changed to show hourly on call and isn't adversely affecting the method.
screen shot 2017-07-28 at 11 52 32 am

File: app/views/assignments/_current.haml (hunk 22 above)
Comment (jforgue) [inline app/views/assignments/_current.haml]: Maybe the person on call makes a big mistake? Then they're only vaguely successful at their assignment, right?

File: app/views/layouts/_nav.haml (hunk 28 above)
Comment (jforgue) [inline app/views/layouts/_nav.haml]: Ah, I wasn't being very observant when I made that comment. As someone less experienced with haml and well, code in general, it made it more readable to me. I see what you mean now though. I'll make sure to keep it consistent with the pros. 👍

File: app/views/assignments/_current.haml (hunk 22 above)
Comment (mboneil10) [inline app/views/assignments/_current.haml]: Lol

File: spec/features/assignments/index_spec.rb [hunk 32]
```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
```
Comment (mboneil10) [inline spec/features/assignments/index_spec.rb]: WHOA

Comment (mboneil10) [inline spec/features/assignments/index_spec.rb]: Not in a bad way. I'm impressed with how much you've done in the test suite. Does it all pass?

File: spec/features/assignments/index_spec.rb [hunk 33]
```diff
@@ -0,0 +1,63 @@
...
+    set_current_user(user)
+  end
+
+  it 'highlights today' do
+    visit roster_assignments_url(roster)
+    expect(page).to have_selector('td.cal-cell.current-day', text: '14')
+  end
+  context 'assignment belongs to user' do
+    it 'appears highlighted yellow for your assignment' do
+      create :assignment, start_date: 2.days.ago, end_date: 2.days.since,
```
Comment (mboneil10) [inline spec/features/assignments/index_spec.rb]: Oh, that's nifty! `2.days.since`.

File: spec/features/assignments/index_spec.rb (hunk 32 above)
Comment (jforgue) [inline spec/features/assignments/index_spec.rb]: Yes, thanks to a lot of guidance.

File: spec/features/assignments/index_spec.rb (hunk 33 above)
Comment (jforgue) [inline spec/features/assignments/index_spec.rb]: Yeah, Timecop is really cool!

Comment (dfaulken) [inline spec/features/assignments/index_spec.rb]: The `2.days.since` syntax is actually just ActiveSupport. But we can use it and always know how it will behave thanks to Timecop :-)

File: spec/features/assignments/index_spec.rb [hunk 34]
```diff
@@ -0,0 +1,63 @@
...
+                          user: user, roster: roster
+      visit roster_assignments_url(roster)
+      expect(page).to have_selector('td .cal-event.assignment-user', count: 5)
+    end
+  end
+
+  context 'assignment does not belong to user' do
+    it 'appears highlighted blue for other assignments' do
```
Comment (mboneil10) [inline spec/features/assignments/index_spec.rb]: Should `it` say 'highlighted blue'? As far as I can tell, the other assignments are highlighted differently than the current user's assignments by `assignment-user` versus `assignment`. Nothing about the color. Just something I noticed.

File: spec/features/assignments/index_spec.rb [hunk 35]
```diff
@@ -3,7 +3,61 @@
...
+      create :assignment, start_date: 2.days.ago, end_date: 2.days.since,
+                          user: roster_user(roster), roster: roster
+      visit roster_assignments_url(roster)
+      expect(page).to have_selector('td .cal-event.assignment', count: 5)
+    end
+  end
+
+  context 'start of assignment' do
+    it 'has a left radius and a smaller width' do
```
Comment (mboneil10) [inline spec/features/assignments/index_spec.rb]: Smaller width than what exactly?

File: spec/features/assignments/index_spec.rb (hunk 34 above)
Comment (jforgue) [inline spec/features/assignments/index_spec.rb]: You make a good point, I change it before pushing the rest of the tests.

File: spec/features/assignments/index_spec.rb [hunk 36]
```diff
@@ -10,54 +10,65 @@
...
+
   it 'highlights today' do
     visit roster_assignments_url(roster)
     expect(page).to have_selector('td.cal-cell.current-day', text: '14')
   end
   context 'assignment belongs to user' do
-    it 'appears highlighted yellow for your assignment' do
-      create :assignment, start_date: 2.days.ago, end_date: 2.days.since,
+    it 'appears highlighted for your assignment' do
+      create :assignment, start_date: 3.days.ago, end_date: 3.days.since,
```
Comment (mboneil10) [inline spec/features/assignments/index_spec.rb]: Why did you make this change? Just to have a week's worth of an assignment?

File: spec/features/assignments/index_spec.rb [hunk 37]
```diff
@@ -10,54 +10,65 @@
...
       visit roster_assignments_url(roster)
-      expect(page).to have_selector('td .cal-event.assignment-user', count: 5)
+      expect(page).to have_selector('td .cal-event.assignment-user', count: 7)
     end
   end
 
   context 'assignment does not belong to user' do
-    it 'appears highlighted blue for other assignments' do
-      create :assignment, start_date: 2.days.ago, end_date: 2.days.since,
+    it 'appears highlighted differently for other assignments' do
```
Comment (mboneil10) [inline spec/features/assignments/index_spec.rb]: Nice

File: spec/features/assignments/index_spec.rb [hunk 38]
```diff
@@ -10,54 +10,65 @@
...
                           user: user, roster: roster
       visit roster_assignments_url(roster)
       expect(page).to have_selector('td .cal-event.assignment-start', count: 1)
     end
   end
 
   context 'end of assignment' do
-    it 'has a right radius' do
-      create :assignment, start_date: 2.days.ago, end_date: 2.days.since,
+    it 'has a right radius and a smaller width than the cell' do
```
Comment (mboneil10) [inline spec/features/assignments/index_spec.rb]: I can't tell that the end of the assignment has a smaller width from the test. You've moved this from line 36 to line 49. Just wanted to make sure you did that on purpose.

File: spec/features/assignments/index_spec.rb [hunk 39]
```diff
@@ -10,54 +10,65 @@
...
+    it 'has a radius around the day and a smaller width than the cell' do
       create :assignment, start_date: Date.today, end_date: Date.today,
                           user: user, roster: roster
       visit roster_assignments_url(roster)
       expect(page).to have_selector('td .cal-event.assignment-only.width',
                                     count: 1)
     end
   end
+
+  context 'active page high lighted in the nav bar' do
```
Comment (mboneil10) [inline spec/features/assignments/index_spec.rb]: Think it should be _highlighted_. 👍

File: spec/features/assignments/index_spec.rb [hunk 40]
```diff
@@ -10,54 +10,65 @@
...
+    it 'applies active class to current tab in nav-bar' do
+      visit roster_assignments_url(roster)
+      expect(page).to have_selector('nav li.active', count: 1)
+    end
```
Comment (mboneil10) [inline spec/features/assignments/index_spec.rb]: Great!

File: spec/features/assignments/index_spec.rb (hunk 38 above)
Comment (jforgue) [inline spec/features/assignments/index_spec.rb]: Right. Yeah I had it on the wrong line before.

File: spec/features/assignments/index_spec.rb (hunk 36 above)
Comment (jforgue) [inline spec/features/assignments/index_spec.rb]: Yes, I wanted it to be a more realistic scenario.

File: app/helpers/assignment_index_helper.rb [hunk 41]
```diff
@@ -7,17 +7,25 @@ def cal_day(day:)
 
   def cal_assignment(assignment:, day:)
     return if assignment.blank?
-    assign = 'assignment'
-    assign += '-user' if assignment.user == @current_user
+    [base_assignment_class(assignment),
```
Comment (mboneil10) [inline app/helpers/assignment_index_helper.rb]: Yesss! This is great!

File: app/helpers/assignment_index_helper.rb [hunk 42]
```diff
@@ -7,17 +7,25 @@ def cal_day(day:)
...
-    else
-      assign
+  def width_assignment_class(assignment, date)
+    if date == assignment.start_date && date == assignment.end_date
+      'assignment-only width'
+    elsif date == assignment.start_date
+      'assignment-start'
+    elsif date == assignment.end_date
+      'assignment-end width'
+    else ''
```
Comment (mboneil10) [inline app/helpers/assignment_index_helper.rb]: Okay, looks good
